File: FilterPositiveThenAverage.py
# Filter Positive Then Average
# - Given a list of integers, filter only positive numbers and calculate their average.
# - If there are no positive numbers, return 0.
# - Time: O(n), Space: O(n)

def filterPositiveThenAverage(nums: list[int]) -> float:
    positives = [x for x in nums if x > 0]
    return sum(positives) / len(positives) if positives else 0

File: FilterPositiveThenAverage.cs
// FilterPositiveThenAverage
// - Given a list of integers, first filter out only positive numbers.
// - Then calculate and return the average of those positive numbers.
// - If there are no positive numbers, return 0.

using System;
using System.Linq;

class Program
{
    static double FilterPositiveThenAverage(int[] nums)
    {
        var positives = nums.Where(x => x > 0).ToList();
        return positives.Count > 0 ? positives.Average() : 0;
    }
}
